app/controllers/event_notes_controller.rb

```diff
@@ -0,0 +1,82 @@
+class EventNotesController < ApplicationController
+  include SerializeDataHelper
+
+  rescue_from Exceptions::EducatorNotAuthorized, with: :redirect_unauthorized!
+
+  before_action :authorize!, except: [:names]
+
+  def authorize!
+    student = Student.find(params[:student_id])
+    educator = current_educator
+    raise Exceptions::EducatorNotAuthorized unless educator.is_authorized_for_student(student)
+  end
+
+  def create
+    clean_params = params.require(:event_note).permit(*[
+      :student_id,
+      :event_note_type_id,
+      :text
+    ])
+
+    event_note = EventNote.new(clean_params.merge({
+      educator_id: current_educator.id,
+      recorded_at: Time.now
```

Comment:
It looks like on first creation, the even't note's `recorded_at` will be the same as the `created_at` and `updated_at` values. If an event note revision is created, you set a `recorded_at value` there to match event note's value and then updated the event_note's value for `recorded_at`. I think that this behavior could be simplified by removing the `recorded_at` columns and referring to the models' `updated_at` columns wherever you need that information. You can replicate the behavior of updating the event note's timestamp when an event note revision is created by calling `event_note.touch` http://apidock.com/rails/ActiveRecord/Timestamp/touch